storage/file.go

```diff
@@ -1,118 +1,54 @@
 package storage
 
 import (
-	"encoding/xml"
 	"errors"
 	"fmt"
 	"io"
 	"net/http"
 	"net/url"
 	"strconv"
-	"strings"
 )
 
-// FileServiceClient contains operations for Microsoft Azure File Service.
-type FileServiceClient struct {
-	client Client
-}
-
-// A Share is an entry in ShareListResponse.
-type Share struct {
-	Name       string          `xml:"Name"`
-	Properties ShareProperties `xml:"Properties"`
-}
+const fourMB = uint64(4194304)
+const oneTB = uint64(1099511627776)
 
-// A Directory is an entry in DirsAndFilesListResponse.
-type Directory struct {
-	Name string `xml:"Name"`
-}
-
-// A File is an entry in DirsAndFilesListResponse.
+// File represents a file on a share.
 type File struct {
-	Name       string         `xml:"Name"`
+	fsc        *FileServiceClient
+	Metadata   map[string]string
+	Name       string `xml:"Name"`
+	parent     *Directory
 	Properties FileProperties `xml:"Properties"`
```

Comment:
Unfortunately this breaks retrieval of file properties, see the call to headerFromStruct() in SetProperties().  We could probably come up with a solution if we want to make this change however I think we can look at that separately.